consensus/src/persistent_liveness_storage.rs

```diff
@@ -407,10 +407,7 @@ impl<T: Payload> PersistentLivenessStorage<T> for StorageWriteProxy {
     }
 
     fn retrieve_validator_change_proof(&self, version: u64) -> Result<ValidatorChangeProof> {
-        let (latest_lis, mut proofs, _) = self.libra_db.get_state_proof(version)?;
-        if latest_lis.ledger_info().version() == version {
-            proofs.ledger_info_with_sigs.push(latest_lis);
-        }
+        let (_, proofs, _) = self.libra_db.get_state_proof(version)?;
```

Comment:
My Pr is about to conflict with your PR. This is actually buggy, since there is a possibility that my version isn't as new as the last epoch change...